lib/Conversion/ImportVerilog/TimingControls.cpp

```diff
@@ -87,6 +104,41 @@ struct DelayControlVisitor {
   }
 };
 
+struct LTLClockControlVisitor {
+  Context &context;
+  Location loc;
+  OpBuilder &builder;
+  Value seqOrPro;
+
+  Value visit(const slang::ast::SignalEventControl &ctrl) {
+    auto edge = convertEdgeKindLTL(ctrl.edge);
+    auto expr = context.convertRvalueExpression(ctrl.expr);
+    if (!expr) {
+      mlir::emitError(loc, "failed to convert expression");
+      return Value{};
+    }
+    Value condition;
+    if (ctrl.iffCondition) {
+      condition = context.convertRvalueExpression(*ctrl.iffCondition);
+      condition = context.convertToBool(condition, Domain::TwoValued);
+      if (!condition) {
+        mlir::emitError(loc, "failed to convert iff condition");
+        return Value{};
+      }
+    }
+    expr = context.convertToBool(expr, Domain::TwoValued);
+    expr.setType(builder.getI1Type());
```

Comment:
Forcing the type does not always work. It implicitly relies on the value being produced by an operation that can have its return type trivially changed. This works for `moore.conversion`, but `convertToBool` might not insert a conversion op at all if it isn't necessary. Your `convertToI1` helper would be an excellent thing to use here!